cmd/nodecmd/loadtestStop.go

```diff
@@ -94,6 +94,37 @@ func stopLoadTest(_ *cobra.Command, args []string) error {
 		return err
 	}
 	removedLoadTestHosts := []*models.Host{}
+	if len(loadTestsToStop) == 0 {
+		return fmt.Errorf("no load test instances to stop in cluster %s", clusterName)
+	}
+	existingLoadTestInstance, err := getExistingLoadTestInstance(clusterName, loadTestsToStop[0])
+	if err != nil {
+		return err
+	}
+	nodeToStopConfig, err := app.LoadClusterNodeConfig(existingLoadTestInstance)
+	if err != nil {
+		return err
+	}
+	clusterNodes, err := getClusterNodes(clusterName)
+	if err != nil {
+		return err
+	}
+	cloudSecurityGroupList, err := getCloudSecurityGroupList(clusterNodes)
+	if err != nil {
+		return err
+	}
+	filteredSGList := utils.Filter(cloudSecurityGroupList, func(sg regionSecurityGroup) bool { return sg.cloud == nodeToStopConfig.CloudService })
+	if len(filteredSGList) == 0 {
+		return fmt.Errorf("no endpoint found in the  %s", nodeToStopConfig.CloudService)
+	}
+	ec2SvcMap := make(map[string]*awsAPI.AwsCloud)
+	for _, sg := range filteredSGList {
+		sgEc2Svc, err := awsAPI.NewAwsCloud(awsProfile, sg.region)
```

Comment:
can you first check if the region is already present in the map?

Reply:
getCloudSecurityGroupList checks for diff security group name also, so there is a chance that a region is repeated